Add activity date filters to the students report screen

m1r0 opened the pull request:
Part of #4833

### Changes proposed in this Pull Request

* Add "start date" and "end date" filters to the students report screen.

### Testing instructions

* Have multiple users that have course "last activity" on different days.
* Go to Sensei LMS -> Reports -> Students.
* Make sure the start date is set to 30 days ago by default.
* Play around with the start/end date filters and make sure both filters are working.
* Try to do an export while using the filters.
* Make sure the search is working.

### Screenshot

<img width="1286" alt="Screenshot on 2022-03-11 at 03-35-01" src="https://user-images.githubusercontent.com/1612178/157784941-9e9f3de8-8762-43c5-af45-4c502fca3814.png">

donnapep commented:
This one seemed...fun. 😅 

I believe that we wanted to set the default date range to the last 30 days so that we improve performance on this page a bit by not fetching all the rows at once.

It's also not obvious that _Start date_ and _End date_ are searching _Last Activity_. I think we may need to specify that in the labels.

m1r0 commented:
> I believe that we wanted to set the default date range to the last 30 days so that we improve performance on this page a bit by not fetching all the rows at once.

Totally forgot about that requirement. 😅 Do you still want to set the filter if the site has X number of students? That might turn out to be a problem because `get_learners` is called after the filter is rendered, so we would need to call it twice and cache the results on the first call. Another option would be to use the site users count `count_users()` or just set the 30 days range by default no matter the students count.

> It's also not obvious that Start date and End date are searching Last Activity. I think we may need to specify that in the labels.

Yeah, I agree. I'm not sure about the design, so what do you think about something like:

<img width="368" alt="Screenshot on 2022-03-10 at 04-19-00" src="https://user-images.githubusercontent.com/1612178/157575409-6cb0e416-b697-4a63-9fa2-2d350705203b.png">

donnapep commented:
> just set the 30 days range by default no matter the students count.

This one sounds good to me. The X student threshold was more applicable when I thought we'd be filtering in the same way as we're filtering lessons.

> Yeah, I agree. I'm not sure about the design, so what do you think about something like:

<img width="368" alt="Screenshot on 2022-03-10 at 04-19-00" src="https://user-images.githubusercontent.com/1612178/157575409-6cb0e416-b697-4a63-9fa2-2d350705203b.png">

LGTM!

m1r0 commented:
> just set the 30 days range by default no matter the students count.

Added in 0b323a861e7cc43f3bd5622ca6e1acb811dd8302.

Also updated the design and added the label in 16b548b1c78a2a90def785dd4cb6f13aaeba6e15.

donnapep reviewed (approved):
👍🏻
